### PR TITLE
Incorporate JaxLinOp with GPJax

### PR DESCRIPTION
**Pull request by Daniel-Dodd**
This PR incorporates `JaxLinOp` with `GPJax`. 

(Still need to update documentation).

### CODE REVIEW

File: gpjax/gps.py [hunk 1]
```diff
@@ -789,7 +778,7 @@ def mll(params: Dict):
             wx = params["latent"]
 
             # f(x) = μx  +  Lx wx
-            fx = μx + jnp.matmul(Lx, wx)
+            fx = μx + Lx @ wx
```
Comment (thomaspinder) [inline gpjax/gps.py:781]: I understand why we can't right now but it'd be cleaner if we didn't have to use `@` for matmuls.

File: gpjax/variational_families.py [hunk 2]
```diff
@@ -22,12 +22,15 @@
 from chex import dataclass
 from jaxtyping import Array, Float
 
+from jaxlinop import identity as I
```
Comment (thomaspinder) [inline gpjax/variational_families.py:25]: Same comment as above re. `identitity` and `I`

**Review by thomaspinder — CHANGES_REQUESTED** [review]
Looks good to me. A few minor comments and then we can merge.

File: gpjax/gps.py (hunk 1 above)
Comment (Daniel-Dodd) [reply gpjax/gps.py:781]: An issue is open for this on `JaxLinOp` (JaxGaussianProcesses/JaxLinOp/#2).

**Review by thomaspinder — APPROVED** [review]

**Pull request merged**